jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunExecutorActor.scala

```diff
@@ -82,13 +89,26 @@ class JobRunExecutorActor(
     context.become(starting)
   }
 
+  def scheduleTimeout(): Unit = {
+    import scala.concurrent.duration._
+
+    jobRun.startingDeadline.foreach { deadline =>
+      val now = clock.now()
+      val from = run.createdAt.plus(deadline.toMillis)
+      val timeout = Seconds.secondsBetween(now, from).getSeconds.seconds
```

Comment:
this timeout could be nonpositive...  imagine a task that was scheduled, metronome goes now and comes up after the deadline.   
For timeout <= 0 
1. we do not want to schedule anything
2. we do want to call `cancelStartingDeadline`